jsonrpc2.go

```diff
@@ -226,7 +226,10 @@ const (
 
 // Handler handles JSON-RPC requests and notifications.
 type Handler interface {
-	// Handle is called to handle a request.
+	// Handle is called to handle a request. No other requests are handled
+	// until it returns. If you do not require strict ordering behaviour
+	// of received RPCs, it is suggested to wrap your handler in
+	// AsyncHandler.
```

Comment:
> No other requests are handled until it returns.

That means until the function returns, not until there is a response. See the signature for the handler, it does not return a response.